lib/src/configuration.dart

```diff
@@ -47,29 +48,45 @@ typedef InitialDataCallback = void Function(Realm realm);
 /// Configuration used to create a [Realm] instance
 /// {@category Configuration}
 abstract class Configuration {
-  static const String _defaultRealmName = 'default.realm';
 
-  static String get _defaultStorageFolder {
-    if (Platform.isAndroid || Platform.isIOS) {
-      return realmCore.getFilesPath();
+  /// The default realm filename to be used. 
+  static String get defaultRealmName => _path.basename(defaultRealmPath);
+  static set defaultRealmName(String name) => defaultRealmPath = _path.join(_path.dirname(defaultRealmPath), _path.basename(name));
+  
+
+  /// The platform dependent path used to store realm files
+  ///
+  /// On Flutter Android and iOS this is the application's data directory.
+  /// On Flutter Windows this is the `C:\Users\username\AppData\Roaming\app_name` directory.
+  /// On Flutter macOS this is the `/Users/username/Library/Containers/app_name/Data/Library/Application Support` directory.
+  /// On Flutter Linux this is the `/home/username/.local/share/app_name` directory.
+  /// On Dart standalone Windows, macOS and Linux this is the current directory.
+  static String get defaultStoragePath {
```

Comment:
I suggest we use `Directory` here instead of `String`

Reply:
what's the benefit. having it as a string makes it easy to combine with other strings. also the others properties are Strings and not `File`

Reply:
The benefit being that the user needs to pass a valid path.

Reply:
This is a getter. There is no way for the user to pass a value here. Also on other places we use Strings for the paths